int32 contrast
---
int32 error